src/HeckeMoreStuff.jl

```diff
@@ -391,11 +391,19 @@ end
 
 Tests if $a$ is nilpotent.
 """
-function is_nilpotent(a::ResElem{T}) where {T<:IntegerUnion}
-  #a is nilpontent if it is divisible by all primes divising the modulus
-  # the largest exponent a prime can divide is nbits(m)
-  l = nbits(modulus(a))
-  return iszero(a^l)
+function is_nilpotent(res::ResElem{T}) where {T<:IntegerUnion}
+  # Code below is faster than
+  #    m=modulus(res);
+  #    return powermod(lift(res),nbits(m),m)==0
+  m = modulus(res)
+  r = lift(res)
+  while true
+    g = gcd(r, m)
+    (g == m) && return true
+    (g == 1) && return false
+    m /= g
```

Comment:
```suggestion
    m = divexact(m, g)
```
Since e.g. `/` is doing the wrong thing if `T === Int`:
```
julia> Int(2)/Int(3)
0.6666666666666666
```

Reply:
Bother!  I was hoping to do "in place" arithmetic with `m /= g` but I suspect that `m = divexact(m,g)` is not in place.  Too bad, I guess.  I'll check what the time penalty is...

Reply:
`m /= g` is just syntactic sugar for `m = m / g` and does not work in place anyway.

We have `divexact!` for a few types to support in-place, but AFAIK this is still mostly in Nemo and and not yet generically implemented in AA. That could be subjective of a future PR. For now just use `divexact` ?

Reply:
**Question:**  in `AbstractAlgebra/src/Residue.jl` near line 73 there is the function definition
```
lift(a::ResElem{Int}) = BigInt(data(a))
```
Why is the result cast to a `BigInt`?  I had expected the result to be of type `Int` or maybe `UInt`; and if the result is to be cast to a "big integer" then wouldn't `ZZRingElem` be more natural (but then it seems that `ZZRingElem` is not visible inside `AbstractAlgebra`... sigh!)

Reply:
`divexact` was faster -- good!  I shall replace the call to `lift` by a call to `data` so that the code actually does what the reader might reasonably expect.  I shall also add an overflow test.

Reply:
I've pushed all the changes... waiting for GitHub to find my mistakes...